llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -52,6 +52,11 @@ static cl::opt<bool> ForceEmitZeroFlag(
   cl::desc("Force all waitcnt instrs to be emitted as s_waitcnt vmcnt(0) expcnt(0) lgkmcnt(0)"),
   cl::init(false), cl::Hidden);
 
+static cl::opt<bool>
+    PreciseMemOpFlag("amdgpu-precise-memory-op",
+                     cl::desc("Emit s_waitcnt 0 after each memory operation"),
+                     cl::init(false));
+
```

Comment:
I think this should be fused into an enum flag with the existing waitcnt flag.

Reply:
Sure. I will look into this.

Reply:
Adding a command-line option for llc is from the 1st version of this patch. Now that we have decided to add a clang command-line option, this llc option is not really necessary anymore.